refinery/ui/source/js/dashboard/controllers/analysis-deletion.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Created by scott on 9/28/16.
+ */
+
+'use strict';
+
+function AnalysisDeleteCtrl (
+  $log,
+  $uibModalInstance,
+  _,
+  deletionService,
+  config,
+  analysis,
+  analyses,
+  analysesReloadService,
+  isOwner,
+  dashboardDataSetsReloadService
+) {
+  this.$log = $log;
+  this._ = _;
+  this.config = config;
+  this.analysis = analysis;
+  this.analyses = analyses;
+  this.$uibModalInstance = $uibModalInstance;
+  this.deletionService = deletionService;
+  this.analysesReloadService = analysesReloadService;
+  this.isOwner = isOwner;
+  this.dashboardDataSetsReloadService = dashboardDataSetsReloadService;
+}
+
+/**
+ * Cancel object deletion.
+ * @type  {function}
+ */
+AnalysisDeleteCtrl.prototype.cancel = function () {
+  this.deletionMessage = null;
+  this.$uibModalInstance.dismiss('cancel');
+};
+
+/**
+ * Delete an Analysis
+ * @type   {function}
+ */
+AnalysisDeleteCtrl.prototype.delete = function () {
+  var that = this;
+  that.isDeleting = true;
+  that.deletionMessage = null;
+  that.deleteSuccessful = false;
+
+  this
+    .deletionService
+    .delete({
+      model: this.config.model,
+      uuid: this.config.uuid
+    })
+    .$promise
+    .then(function (response) {
```

Comment:
then has a built in error callback. See docs, https://docs.angularjs.org/api/ng/service/$q

```then(successCallback, [errorCallback], [notifyCallback]) – regardless of when the promise was or will be resolved or rejected, then calls one of the success or error callbacks asynchronously as soon as the result is available. The callbacks are called with a single argument: the result or rejection reason. Additionally, the notify callback may be called zero or more times to provide a progress indication, before the promise is resolved or rejected.

This method returns a new promise which is resolved or rejected via the return value of the successCallback, errorCallback (unless that value is a promise, in which case it is resolved with the value which is resolved in that promise using promise chaining). It also notifies via the return value of the notifyCallback method. The promise cannot be resolved or rejected from the notifyCallback method. The errorCallback and notifyCallback arguments are optional.

catch(errorCallback) – shorthand for promise.then(null, errorCallback)

finally(callback, notifyCallback) – allows you to observe either the fulfillment or rejection of a promise, but to do so without modifying the final value. This is useful to release resources or do some clean-up that needs to be done whether the promise was rejected or resolved. See the full specification for more information.``` @